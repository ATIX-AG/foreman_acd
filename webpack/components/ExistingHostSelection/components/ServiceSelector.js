import React from 'react';
import PropTypes from 'prop-types';
import ExtSelect from '../../common/ExtSelect';
import RailsData from '../../common/RailsData'

const ServiceSelector= ({
  label,
  hidden,
  viewText,
  selectValue,
  onChange,
  options,
  additionalData,
}) =>{
  return (
    <div className="form-group">
      <label className="col-md-2 control-label">{label}</label>
      <div className="col-md-4">
        <ExtSelect
          hidden={hidden}
          editable={true}
          viewText={viewText}
          selectValue={selectValue}
          onChange={onChange}
          options={options}
          additionalData={additionalData}
        />
        <RailsData
          key='service_id'
          view='existing_host_selection'
          parameter='service_id'
          value={selectValue}
        />
      </div>
    </div>
  );
};

ServiceSelector.propTypes = {
  label: PropTypes.string.isRequired,
  viewText: PropTypes.string,
  selectValue: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.object,
  additionalData: PropTypes.object,
};

export default ServiceSelector;
