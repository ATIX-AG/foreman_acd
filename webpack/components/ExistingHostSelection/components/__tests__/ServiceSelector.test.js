import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import ServiceSelector from '../ServiceSelector';

const noop = () => {};

const data = [
  { name: 'report 1', status: 'done' },
  { name: 'report 2', status: 'pending' }
];

const fixtures = {
  'should render the service selector': {
    hidden: false,
    label: 'Test Label',
    viewText: 'view service',
    selectValue: '1',
    onChange: noop,
    options: { first: 'first', second: 'second'},
    additionalData: { moreData: 'moooore' },
  },
  'should render the hidden service selector': {
    hidden: true,
    label: 'Test Label',
    viewText: 'view service',
    selectValue: '1',
    onChange: noop,
    options: { first: 'first', second: 'second'},
    additionalData: { },
  },
};

describe('ServiceSelector', () =>
  testComponentSnapshotsWithFixtures(ServiceSelector, fixtures));

