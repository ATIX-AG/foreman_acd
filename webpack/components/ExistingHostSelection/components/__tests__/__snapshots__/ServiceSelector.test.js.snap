// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceSelector should render the hidden service selector 1`] = `
<div
  className="form-group"
>
  <label
    className="col-md-2 control-label"
  >
    Test Label
  </label>
  <div
    className="col-md-4"
  >
    <ExtSelect
      additionalData={Object {}}
      editable={true}
      hidden={true}
      onChange={[Function]}
      options={
        Object {
          "first": "first",
          "second": "second",
        }
      }
      selectValue="1"
      viewText="view service"
    />
    <RailsData
      key="service_id"
      parameter="service_id"
      value="1"
      view="existing_host_selection"
    />
  </div>
</div>
`;

exports[`ServiceSelector should render the service selector 1`] = `
<div
  className="form-group"
>
  <label
    className="col-md-2 control-label"
  >
    Test Label
  </label>
  <div
    className="col-md-4"
  >
    <ExtSelect
      additionalData={
        Object {
          "moreData": "moooore",
        }
      }
      editable={true}
      hidden={false}
      onChange={[Function]}
      options={
        Object {
          "first": "first",
          "second": "second",
        }
      }
      selectValue="1"
      viewText="view service"
    />
    <RailsData
      key="service_id"
      parameter="service_id"
      value="1"
      view="existing_host_selection"
    />
  </div>
</div>
`;
